v2/internal/middleware/middleware.go

```diff
@@ -31,15 +33,15 @@ func (r *stack[IN, OUT]) AddMiddleware(mw Middleware[IN, OUT]) {
 	r.middlewares = append(r.middlewares, mw)
 }
 
-func (r *stack[IN, OUT]) Process(options IN) OUT {
-	var next func(item IN) OUT = nil
+func (r *stack[IN, OUT]) Process(ctx context.Context, options IN) OUT {
+	var next func(c context.Context, item IN) OUT = nil
 	middlewares := append(r.middlewares)
-	next = func(item IN) OUT {
+	next = func(c context.Context, item IN) OUT {
 		nextMw := middlewares[0]
 		middlewares = middlewares[1:]
-		return nextMw.Process(item, next)
+		return nextMw.Process(c, item, next)
 	}
-	return next(options)
+	return next(ctx, options)
```

Comment:
minor: it's a bit hard to read when everything is named as next, it's probably better to rename `nextMw` to `currentMw`

Reply:
renamed :) thanks.